packages/ebsi-authorization-client/src/agent/EBSIAuthorizationClient.ts

```diff
@@ -0,0 +1,363 @@
+        method: 'POST',
+        headers: new Headers({
+          ContentType: 'application/x-www-form-urlencoded',
+          Accept: 'application/json',
+        }),
+        body: formData,
+      })
+    ).json()
+  }
+
+  private async createOAuth2Session(args: CreateOAuth2SessionArgs): Promise<CreateOAuth2SessionResponse> {
+    return await (
+      await fetch(`${this.configuration.issuer}/oauth2-session`, {
+        method: 'POST',
+        headers: new Headers({
+          ContentType: 'application/json',
+          Accept: 'application/json',
+        }),
+        body: JSON.stringify(args),
+      })
+    ).json()
+  }
+}
+
+export async function createVPJwt(
+  args: {
+    keyPair: { publicKeyHex: string; privateKeyHex: string }
+    did: string
+    vc: string
+    audience: string
+    domain: string
+    nbf?: number
+    exp?: number
+    nonce?: number
+  },
+  context: IRequiredContext,
+): Promise<string> {
+  const { keyPair, did, vc, audience, nbf, exp, nonce } = args
+
+  if (!keyPair || !keyPair.privateKeyHex || !keyPair.publicKeyHex) {
+    throw new Error(`Private key is required`)
+  }
+
+  if (!args.vc) {
+    throw new Error('Verifiable Credential not defined')
+  }
+
+  const privateKeyJWT = await exportJWK(u8a.fromString(keyPair.privateKeyHex, 'hex'))
+  const publicKeyJWT = await exportJWK(u8a.fromString(keyPair.publicKeyHex, 'hex'))
+
+  let verifiableCredential: string[]
+
+  if (vc === 'empty') {
+    verifiableCredential = []
+  } else if (Array.isArray(vc)) {
+    verifiableCredential = vc
+  } else {
+    verifiableCredential = [vc]
+  }
+
+  const resolvedDid = await context.agent.didManagerGet({ did })
+  if (!resolvedDid) {
+    throw new Error(`${did} could not be resolved`)
+  }
+
+  const payload = {
+    id: `urn:did:${uuid()}`,
+    '@context': ['https://www.w3.org/2018/credentials/v1'],
+    type: ['VerifiablePresentation'],
+    holder: did,
+    verifiableCredential,
+  }
+
+  const vpJwtPayload = {
+    holder: did,
+    jti: typeof payload['id'] === 'string' ? payload['id'] : '',
+    sub: did,
+    iss: did,
+    ...(nbf && { nbf }),
+    ...(exp && { exp }),
+    iat: Math.floor(Date.now() / 1000),
+    aud: audience,
+    vp: payload,
+    nonce: nonce ?? uuid(),
+    exp: Math.floor(Date.now() / 1000) + 900,
+    nbf: Math.floor(Date.now() / 1000) - 100,
+  }
+
+  const jwtHeader: JWTHeader = {
+    typ: 'JWT',
+    kid: privateKeyJWT.kid,
+    alg: privateKeyJWT.alg ?? 'ES256',
+  }
+
+  const holderKey: JsonWebKey2020 = {
+    id: privateKeyJWT.kid!,
+    type: 'JsonWebKey2020',
+    controller: did,
+    publicKeyJwk: publicKeyJWT,
+    privateKeyJwk: privateKeyJWT,
+  }
+
+  const signingKey = await JsonWebKey.from(holderKey)
+  const vpKey = await JsonWebKey.from(
+    await signingKey.export({
+      type: 'JsonWebKey2020',
+      privateKey: true,
+    }),
+    {
+      detached: false,
+      header: jwtHeader,
+    },
+  )
+  const vpSigner = vpKey.signer()
```

Comment:
Don;t use external libs for this. The agent is perfectly capable of creating VCs and VPs of course!

Reply:
Removed third-party libraries and signing JWTs with the agent now